### PR TITLE
Adding support for a new debugger

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatoliy Bazko <abazko@redhat.com>

### What does this PR do?
Describes how to add a new debugger into Eclipse Che.

### What issues does this PR fix or reference?
https://github.com/eclipse/che-docs/issues/719

### CODE REVIEW

File: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc [hunk 1]
```diff
@@ -1,27 +1,31 @@
...
-This paragraph is the procedure module introduction: a short description of the procedure.
+This section describes how to add a new debugger into Eclipse Che.
 
 [discrete]
 == Prerequisites
 
-* A bulleted list of conditions that must be satisfied before the user starts following this assembly.
-* You can also link to other modules or assemblies the user must follow before starting this assembly.
-* Delete the section title and bullets if the assembly has no prerequisites.
+* Install the latest Theia plugin yeoman generator `npm install -g yo @theia/generator-plugin`
```
Comment (rkratky) [inline src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc]:
```suggestion
* Install the latest Che-Theia plug-in Yeoman generator by running:
+
----
$ npm install -g yo @theia/generator-plugin
----
```

File: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
+
 
 [discrete]
 == Procedure
 
-. Start each step with an active verb.
+. Generate sample by running `yo @theia/plugin`. When prompted confirm choices for “Backend plug-in”, “Samples” and then “Debugger sample”. It will create a project and a ready-to-use plugin which can be deployed.
```
Comment (rkratky) [inline src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc]:
```suggestion
. Generate a sample by running `yo @theia/plugin`. When prompted, confirm the choices for *Backend plug-in*, *Samples*, and then *Debugger sample*. This creates a project and a ready-to-use plug-in, which can be deployed.
```

File: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
+
+. Replace the mock implementation of the debug adapter in `src/vscode-mockDebug.ts` and `src/vscode-mockRuntime.ts` by some code that talks to a "real" debugger or runtime.
```
Comment (rkratky) [inline src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc]:
```suggestion
. Replace the mock implementation of the debug adapter in the `src/vscode-mockDebug.ts` and `src/vscode-mockRuntime.ts` files by code that interacts with a real debugger or runtime.
```

File: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
 
-. Include one command or action per step.
+. Build the plugin by running `yarn` in the root of the project.
```
Comment (rkratky) [inline src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc]:
```suggestion
. Build the plug-in by running `yarn` in the root of the project.
```

File: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
+
+. Deploy Che Theia plugin via Che plugin-in registry.
```
Comment (rkratky) [inline src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc]:
```suggestion
. Deploy the Che-Theia plug-in via the plugin-in registry.
```

File: src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc [hunk 6]
```diff
@@ -1,27 +1,31 @@
...
 [discrete]
 == Additional resources
 
-* A bulleted list of links to other material closely related to the contents of the procedure module.
-* For more details on writing procedure modules, see the link:https://github.com/redhat-documentation/modular-docs#modular-documentation-reference-guide[Modular Documentation Reference Guide].
-* Use a consistent system for file names, IDs, and titles. For tips, see _Anchor Names and File Names_ in link:https://github.com/redhat-documentation/modular-docs#modular-documentation-reference-guide[Modular Documentation Reference Guide].
+* link:https://microsoft.github.io/debug-adapter-protocol/[Debug Adapter Protocol]
+* link:https://code.visualstudio.com/api/extension-guides/debugger-extension[VS Code debugger extension documentation]
+* link:https://microsoft.github.io/debug-adapter-protocol/implementors/adapters/[Implementations Debug Adapters]
+* xref:adding-a-che-theia-plug-in-to-the-che-plug-in-registry[Adding a Che-Theia plug-in to the Che plug-in registry]
```
Comment (rkratky) [inline src/main/pages/che-7/hands-on-guidance/proc_adding-support-for-a-new-debugger.adoc]:
```suggestion
* link:adding-a-che-theia-plug-in-to-the-che-plug-in-registry.html[Adding a Che-Theia plug-in to the Che plug-in registry]
```

**Review by rkratky — CHANGES_REQUESTED** [review]

**Comment by tolusha** [comment]
@rkratky I've committed all suggestions.

**Review by rkratky — APPROVED** [review]

**Pull request merged**